packages/nextjs/src/utils/config.ts

```diff
@@ -30,32 +30,50 @@ type EntryPointObject = { import: string | Array<string> };
 const _injectFile = (entryProperty: EntryPropertyObject, injectionPoint: string, injectee: string): void => {
   // can be a string, array of strings, or object whose `import` property is one of those two
   let injectedInto = entryProperty[injectionPoint];
-  // whatever the format, add in the sentry file
-  injectedInto =
-    typeof injectedInto === 'string'
-      ? // string case
-        [injectee, injectedInto]
-      : // not a string, must be an array or object
-      Array.isArray(injectedInto)
-      ? // array case
-        [injectee, ...injectedInto]
-      : // object case
-        {
-          ...injectedInto,
-          import:
-            typeof injectedInto.import === 'string'
-              ? // string case for inner property
-                [injectee, injectedInto.import]
-              : // array case for inner property
-                [injectee, ...injectedInto.import],
-        };
+
+  // Sometimes especially for older next.js versions it happens we don't have an entry point
+  if (injectedInto === undefined) {
+    // eslint-disable-next-line no-console
+    console.error(`[Sentry] Can't inject ${injectee}, no entrypoint is defined.`);
+    return;
+  }
+
+  // In case we inject our client config, we need to add it after the frontend code
+  // otherwise the runtime config isn't loaded. See: https://github.com/getsentry/sentry-javascript/issues/3485
+  const isClient = injectee === './sentry.client.config.js';
+
+  // Object -> Add it
+  if (injectedInto != null && typeof injectedInto == 'object' && !Array.isArray(injectedInto)) {
+    let importVal: string | string[] | EntryPointObject;
+    if (typeof injectedInto.import === 'string') {
+      importVal = isClient ? [injectedInto.import, injectee] : [injectee, injectedInto.import];
+    } else {
+      // If it's not a string, the inner value is an array
+      importVal = isClient ? [...injectedInto.import, injectee] : [injectee, ...injectedInto.import];
+    }
+
+    injectedInto = {
+      ...injectedInto,
+      import: importVal,
+    };
+  }
+
+  // Array -> Add it
+  if (Array.isArray(injectedInto)) {
+    injectedInto = isClient ? [...injectedInto, injectee] : [injectee, ...injectedInto];
+  }
+
+  // String -> We need to make it an array
+  if (typeof injectedInto === 'string') {
+    injectedInto = isClient ? [injectedInto, injectee] : [injectee, injectedInto];
+  }
```

Comment:
Changing the order of the `if`s would require fewer checks (and maybe easier to understand):

```ts
if (typeof injectedInto === 'string') {
  // ...
} else if (Array.isArray(injectedInto)) {
  // ...
} else {
  // We know it's an object at this point, without explicitly checking it
  // ...
}
```